source/common/network/utility.cc

```diff
@@ -732,11 +732,11 @@ Api::IoErrorPtr Utility::readPacketsFromSocket(IoHandle& handle,
               ? (packets_dropped - old_packets_dropped)
               : (packets_dropped + (std::numeric_limits<uint32_t>::max() - old_packets_dropped) +
                  1);
-      ENVOY_LOG_MISC(
-          debug,
-          "Kernel dropped {} datagram(s). Consider increasing receive buffer size and/or "
-          "max datagram size.",
-          delta);
+      ENVOY_BUG(false,
```

Comment:
I don't think this should be an `ENVOY_BUG`, because it's not a bug in envoy. But I think setting the warning level to `warn`, and logging with one of the periodic or other not-every-time frequency log macros makes sense.

Reply:
good point. In fact it is triggered in several integration tests.

Reply:
@danzh2010 if it's triggered in integration tests won't it crash the debug build?  I think we need to sort that out before merging.  Otherwise LGTM
/wait

Reply:
It indeed crashed the build, so I down grade it to warn with ENVOY_LOG_EVERY_POW_2_MISC